File: backend/src/db/entities/recitation-entity.ts
import { Entity, Column, ManyToOne, JoinColumn, OneToOne } from 'typeorm';
import { LocalizedEntity } from './localized-entity';
import { FileEntity } from './file-entity';
import { ReciterEntity } from './reciter-entity';
import { AppBaseEntity } from './base-entity';

export enum RecitationTypes {
  Hafs = 1,
  Shoaba = 2,
}

export const RecitationTypesList = [
  RecitationTypes.Hafs,
  RecitationTypes.Shoaba,
];

@Entity({ name: 'recitation' })
export class RecitationEntity extends AppBaseEntity {
  @Column(type => LocalizedEntity)
  title!: LocalizedEntity;

  @OneToOne(() => FileEntity, {
    eager: true,
  })
  @JoinColumn()
  image!: FileEntity;

  @OneToOne(() => FileEntity, {
    eager: true,
  })
  @JoinColumn()
  audio!: FileEntity;

  @ManyToOne(() => ReciterEntity, {
    eager: true,
  })
  @JoinColumn()
  reciter!: ReciterEntity;

  @Column({
    type: "enum",
    enum: RecitationTypes,
    default: RecitationTypes.Hafs
  })
  recitationType!: RecitationTypes;

  @Column()
  khatmaId!: number;

  @Column({ default: 0 })
  sequence!: number;

  @Column({ default: 0 })
  totalDownloads!: number;

  @Column({ default: 0 })
  totalPlays!: number;

  @Column({ default: 0 })
  duration!: number;
}

File: mobile/lib/src/modules/recitations/domain/entity/recitation_entity.dart
import 'package:kawtharuna/src/core/entity/base/base_entity.dart';
import 'package:kawtharuna/src/core/entity/localization/localized_entity.dart';

class RecitationsListEntity extends BaseEntity {
  final List<RecitationEntity> items;

  const RecitationsListEntity({required this.items});

  @override
  List<Object?> get props => [
        items,
      ];
}

class RecitationEntity extends BaseEntity {
  final int id;
  final int khatmaId;
  final LocalizedEntity? title;
  final String? audio;
  final String? image;
  final int durationInSecond;

  const RecitationEntity({
    required this.id,
    required this.khatmaId,
    this.title,
    this.audio,
    this.image,
    required this.durationInSecond,
  });

  @override
  List<Object?> get props => [
        id,
        khatmaId,
        title,
        audio,
        image,
        durationInSecond,
      ];
}
